File: backend/src/main/java/com/prepunite/company/entity/Company.java
package com.prepunite.company.entity;

import com.prepunite.common.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "companies")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Company extends BaseEntity {

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, unique = true)
    private String slug;

    @Column(name = "logo_url")
    private String logoUrl;

    private String website;

    @Column(columnDefinition = "TEXT")
    private String description;

    private String industry;

    @Column(name = "company_size")
    private String companySize;

    private String headquarters;

    @Column(name = "founded_year")
    private Integer foundedYear;

    @Column(name = "is_active", nullable = false)
    @Builder.Default
    private Boolean isActive = true;

    @OneToMany(mappedBy = "company", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<CompanyRole> roles = new ArrayList<>();

    @OneToMany(mappedBy = "company", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("roundNumber ASC")
    @Builder.Default
    private List<HiringProcess> hiringProcesses = new ArrayList<>();
}

File: frontend/src/types/company.ts
export interface CompanyRole {
  id: string;
  title: string;
  salaryMin?: number;
  salaryMax?: number;
  salaryCurrency?: string;
  eligibility?: string;
  roleType?: string;
  description?: string;
}

export interface HiringProcessRound {
  id: string;
  roundNumber: number;
  roundType: string;
  title: string;
  description?: string;
  durationMinutes?: number;
  tips?: string;
}

export interface Company {
  id: string;
  name: string;
  slug: string;
  logoUrl?: string;
  website?: string;
  description?: string;
  industry?: string;
  companySize?: string;
  headquarters?: string;
  foundedYear?: number;
  isActive: boolean;
  createdAt: string;
  roles?: CompanyRole[];
  hiringProcesses?: HiringProcessRound[];
}

export interface PageResponse<T> {
  content: T[];
  pageable: {
    pageNumber: number;
    pageSize: number;
  };
  totalElements: number;
  totalPages: number;
  last: boolean;
  first: boolean;
}
